rasterio/windows.py

```diff
@@ -303,10 +302,71 @@ def shape(window, height=-1, width=-1):
     values in the window.
     """
     (a, b), (c, d) = evaluate(window, height, width)
-    return b-a, d-c
+    return (b - a, d - c)
 
 
 def window_index(window):
     # "window_" is necessary here to redundancy to disambiguate
     # from transform.get_index and src.index
     return tuple(slice(*w) for w in window)
+
+
+class Window(tuple):
+    """Windows are rectangular subsets of rasters"""
```

Comment:
I think this will cause confusion but maybe only for among users familiar with the original start/stop ranges?  The fact that the window subclasses `tuple` is pretty well obfuscated so new users shouldn't have any idea what's happening behind the scenes unless they try a `__getitem__`.

Rasterio draws from Numpy in many places, and I find the start/stop ranges to be a familiar nod at array slicing.  Their actual representation as nested tuples is a little clunky in practice, but I wonder if a `Window.from_offset()` is more appropriate for GDAL-style constructing?  Letting GDAL's Python binding signatures define our signatures doesn't quite sit right, but if the offset is more intuitive to more users 👍